### ISSUE TITLE
Upstream longitude equality helper from sunpy to astropy

### ISSUE THREAD

**Issue by Prtm2110** [astropy/astropy#16797]
### What is the problem this feature will solve?

Check for the equality/closeness of longitudes 
Reference [3821](https://github.com/sunpy/sunpy/issues/3821)

**Comment by mhvk** [comment]
Is the idea to have this for our tests? Or more general? E.g., one could envisage overriding `np.isclose` and `np.allclose` to take into account wrapping for `Longitude`. Or have an `isclose` method (like we have for `Time`). Not sure what is the best approach.

**Comment by pllim** [comment]
Looks like they want to move https://github.com/sunpy/sunpy/pull/3804 to this repo. 12 lines of code:

```python
from astropy.coordinates import Longitude
from astropy.tests.helper import assert_quantity_allclose
import astropy.units as u


def assert_longitude_allclose(actual, desired, atol=None):
    """
    This works like :func:`~astropy.tests.helper.assert_quantity_allclose`, except it is intended
    for longitude angles (i.e., angles that wrap every 360 degrees).
    """
    difference = Longitude(actual - desired, wrap_angle=180*u.deg)
    assert_quantity_allclose(difference, 0*u.deg, atol=atol)
```

**Comment by nabobalis** [comment]
Hi @pllim and @mhvk. Sorry for the lack of details on the original issue.

Yeah, I was curious if astropy would find some use in this test helper as Albert wrote it to help with some of the coordinate testing.

If there is no desire, then we can keep it in sunpy.

**Comment by eerovaher** [comment]
It would be good to have an `astropy.testing` sub-package that would contain public helper functions for testing, just like how `numpy` has [`numpy.testing`](https://numpy.org/doc/stable/reference/routines.testing.html). `assert_longitude_allclose()` could be one such helper, although it really should allow specifying `rtol` and perhaps also `wrap_angle`.

**Comment by pllim** [comment]
We already have "public helpers" under `astropy.tests`.

**Comment by pllim** [comment]
So, do we have a consensus from `coordinates` sub-package maintainers? 🙏 

@StuartLittlefair @adrn @eerovaher @mhvk

**Comment by mhvk** [comment]
Yes, it could go in `astropy/tests/helper.py` (where there is already `assert_quantity_allclose`). Though it will need a bit of looking around to see where it would actually be used -- we do want to adjust our actual tests if we add this!

**Comment by eerovaher** [comment]
Many of the classes `astropy` provides should be accompanied with public assert helpers, not just `Longitude`, but we can add them one at a time.

**Comment by mhvk** [comment]
> Many of the classes `astropy` provides should be accompanied with public assert helpers, not just `Longitude`, but we can add them one at a time.

Indeed, would be good to (eventually) centralize things like `representation_equal`, etc.

**Comment by adrn** [comment]
Yes I think this would be useful, and as @mhvk said we should see where this could help in some of our coordinates tests.
